lumen/ai/models.py

```diff
@@ -99,3 +100,46 @@ class Topic(BaseModel):
 class VegaLiteSpec(BaseModel):
 
     json_spec: str = Field(description="A vega-lite JSON specification WITHOUT the data field, which will be added automatically.")
+
+
+def make_plan_model(agent_names: tuple[str]):
+    step = create_model(
+        "Step",
+        expert=(Literal[agent_names], FieldInfo(description="The name of the expert to assign a task to.")),
+        instruction=(str, FieldInfo(description="Instructions to the expert to assist in the task."))
+    )
+    return create_model(
+        "Plan",
+        chain_of_thought=(
+            str,
+            FieldInfo(
+                description="Describe at a high-level how the actions of each expert will solve the user query."
+            ),
+        ),
+        steps=(
+            list[step],
+            FieldInfo(
+                description="A list of tuples of the experts name and instructions to that expert to help him solve the overall task"
+            )
+        ),
+    )
```

Comment:
Prefix with `correct_` and mention `refer to your chain_of_thoughts output`. 

```suggestion
    return create_model(
        "Plan",
        chain_of_thought=(
            str,
            FieldInfo(
                description="Describe at a high-level how the actions of each expert will solve the user query."
            ),
        ),
        correct_steps=(
            list[step],
            FieldInfo(
                description="A list of tuples of the experts name and instructions to that expert to help him solve the overall task. Refer to your chain_of_thought output"
            )
        ),
    )
```